fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/CollectJoinConstraint.java

```diff
@@ -92,27 +90,37 @@ public List<Rule> buildRules() {
             }).toRule(RuleType.COLLECT_JOIN_CONSTRAINT),
 
             logicalFilter().thenApply(ctx -> {
-                LeadingHint leading = (LeadingHint) ctx.cascadesContext
-                        .getStatementContext().getHintMap().get("Leading");
-                if (leading == null) {
+                if (!ctx.cascadesContext.isLeadingJoin()) {
                     return ctx.root;
                 }
+                LeadingHint leading = (LeadingHint) ctx.cascadesContext
+                        .getHintMap().get("Leading");
                 LogicalFilter filter = ctx.root;
                 Set<Expression> expressions = filter.getConjuncts();
                 for (Expression expression : expressions) {
-                    Long filterBitMap = calSlotsTableBitMap(leading, expression.getInputSlots(), false);
+                    Long filterBitMap = 0L;
```

Comment:
Pls make sure this part of logic correctness and completeness.

Reply:
done